File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <math.h>
#include <string.h>

int main(void)
{
    // Prompt the user for text
    string s = get_string("Text: ");
    int letter = 0;
    int word = 0;
    int sentence = 0;
    // Calculate number of letters, words and sentences
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (isalpha(s[i]))
        {
            letter++;
        }
        else if (isspace(s[i]))
        {
            word++;
        }
        else if (s[i] == '.' || s[i] == '?' || s[i] == '!')
        {
            sentence++;
        }
    }
    word++;
    // Calculate average no. of letters per 100 words and average number of sentences per 100 words
    float L;
    L = ((float) letter / word) * 100;
    float S;
    S = ((float) sentence / word) * 100;
    // Use the Coleman-Liau formula to calculate grade of the text
    float index;
    index = 0.0588 * L - 0.296 * S - 15.8;
    int X = round(index);
    // Printing the appropriate grade
    if (X < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (X >= 1 && X < 16)
    {
        printf("Grade %i\n", X);
    }
    else if (X >= 16)
    {
        printf("Grade 16+\n");
    }
}

File: readability.py
from cs50 import get_string

# Taking input from user
text = get_string("Text: ")
# Declaring variables to store corresponding count
letters = 0
words = 0
sentences = 0
for i in range(len(text)):
    # Checking for letters
    if text[i].isalpha():
        letters += 1
    # Checking for words
    elif text[i] == " ":
        words += 1
    # Checking for sentences
    elif text[i] in ['?', '!', '.']:
        sentences += 1

# Considering the last word of the sentence
words += 1

# Calculating number of letters and sentences per 100 words
L = (letters / words) * 100
S = (sentences / words) * 100

# Calculating level using Coleman-Liau formula
level = round(0.0588 * L - 0.296 * S - 15.8)

# Printing appropriate level
if level < 1:
    print("Before Grade 1")
elif level > 16:
    print("Grade 16+")
else:
    print(f"Grade {level}")
